src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractChannelInitializer.java

```diff
@@ -73,45 +71,31 @@ protected void initChannel(@NotNull final Channel ch) throws Exception {
         ch.pipeline().addLast(MQTT_MESSAGE_DECODER, new MQTTMessageDecoder(channelDependencies));
         ch.pipeline().addLast(MQTT_MESSAGE_ENCODER, channelDependencies.getMqttMessageEncoder());
         addNoConnectIdleHandler(ch);
+        //MQTT_5_FLOW_CONTROL_HANDLER is added here after CONNECT
         ch.pipeline().addLast(MQTT_MESSAGE_BARRIER, new MessageBarrier(channelDependencies.getMqttServerDisconnector()));
         ch.pipeline().addLast(MQTT_SUBSCRIBE_MESSAGE_BARRIER, new SubscribeMessageBarrier());
```

Comment:
MessageBarrier and SubscribeMessageBarrier should not be permanently in the pipeline, rather they should be added and removed on demand (e.g. added by the SubscribeHandler when receiving a SUBSCRIBE and removing/setting a flag to remove when sending a SUBACK).

Reply:
MessageBarrier is already removing itself after CONNACK